cmake-modules/AzureDoxygen.cmake

```diff
@@ -17,19 +17,33 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_OUTPUT_DIRECTORY docs)
         set(DOXYGEN_LAYOUT_FILE ${CMAKE_SOURCE_DIR}/eng/docs/api/assets/DoxygenLayout.xml)
         set(DOXYGEN_RECURSIVE YES)
+        if (MSVC)
+          set(DOXYGEN_WARN_FORMAT "$file($line) : $text")
+        endif()
+#        set(DOXYGEN_WARN_AS_ERROR FAIL_ON_WARNINGS)
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
         # Setting the INLINE_SOURCES tag to YES will include the body of functions,
         # classes and enums directly into the documentation.
         set(DOXYGEN_INLINE_SOURCES NO)
+        set(DOXYGEN_MARKDOWN_ID_STYLE GITHUB)
         # Skip generating docs for json, test, samples, and private files.
         set(DOXYGEN_EXCLUDE_PATTERNS
             json.hpp
+            package_version.hpp
+            dll_import_export.hpp
```

Comment:
Why do we need to skip these two?

Reply:
Because they contain macros that start with AZURE_<PACKAGE>_, and I don't think we want them in our product documentation. And I don't think we want to have to document the macros that are used for package building.